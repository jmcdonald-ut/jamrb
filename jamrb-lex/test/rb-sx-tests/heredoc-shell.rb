<<`HEREDOC`
cd ~/projects && echo "Get to work!"
HEREDOC

<<~`H`
cd ~/projects && #{foo}
H

<<-`SHELLCODE`
cd ~/projects && #{foo}
SHELLCODE
