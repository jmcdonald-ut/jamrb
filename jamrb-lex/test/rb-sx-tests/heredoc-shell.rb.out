((Token ((1 0) on_heredoc_beg "<<`HEREDOC`"))
 (Token ((1 11) on_nl "\n"))
 (Token ((2 0) on_tstring_content "cd ~/projects && echo \"Get to work!\"\n"))
 (Token ((3 0) on_heredoc_end "HEREDOC\n"))
 (Token ((4 0) on_ignored_nl "\n"))
 (Token ((5 0) on_heredoc_beg "<<~`H`"))
 (Token ((5 6) on_nl "\n"))
 (Token ((6 0) on_tstring_content "cd ~/projects && "))
 (Token ((6 17) on_embexpr_beg "\\#{"))
 (Token ((6 19) on_ident "foo"))
 (Token ((6 22) on_embexpr_end "}"))
 (Token ((6 23) on_tstring_content "\n"))
 (Token ((7 0) on_heredoc_end "H\n"))
 (Token ((8 0) on_ignored_nl "\n"))
 (Token ((9 0) on_heredoc_beg "<<-`SHELLCODE`"))
 (Token ((9 14) on_nl "\n"))
 (Token ((10 0) on_tstring_content "cd ~/projects && "))
 (Token ((10 17) on_embexpr_beg "\\#{"))
 (Token ((10 19) on_ident "foo"))
 (Token ((10 22) on_embexpr_end "}"))
 (Token ((10 23) on_tstring_content "\n"))
 (Token ((11 0) on_heredoc_end "SHELLCODE\n")))
